pytorch_lightning/utilities/cli.py

```diff
@@ -345,9 +458,91 @@ def link_optimizers_and_lr_schedulers(self) -> None:
                 add_class_path = _add_class_path_generator(class_type)
                 self.parser.link_arguments(key, link_to, compute_fn=add_class_path)
 
+    @contextmanager
+    def _prepare_from_registry(self, registry: _Registry):
+        """
+        This context manager is used to simplify unique class instantiation.
+        """
+
+        # find if the users is using shortcut command line.
+        map_user_key_to_info = {}
+        for registered_name, registered_cls in registry.items():
+            for v in sys.argv:
+                if f"={registered_name}" in v:
```

Comment:
I think this comparison has to be more strict. If there is a name registered as `Name` and then an argument is `--*=Name2` this `if` would be true even though `Name2` is not registered.

Reply:
Resolved !